exercise_1/ShortestPath.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by cdd on 18.07.19.
+//
+
+#include "ShortestPath.h"
+#include <iostream>
+#include <algorithm>
+
+ShortestPath::ShortestPath(const std::vector<std::vector<int>>&matrix,const int& topPoint):_matrix(matrix),_topPoint(topPoint) {
+
+}
+
+std::vector<int> ShortestPath::getPaths(std::vector<std::vector<int> > graphMatrix, int topPoint, int start){
+    int  count, index, i, u;
+    bool *visited;
+    std::vector<int> distance(topPoint);
+    visited = new bool [topPoint];
+    for (i=0; i<topPoint; ++i) { distance[i]=std::numeric_limits<int>::max(); visited[i]=false; }
```

Comment:
you may declare 'i' in the for loop
the same in line 22
c++ does not require you to declare all variables at the beginning of the scope